TLM/TLM/UI/MainMenu/MainMenuWindow.ToolPanel.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace TrafficManager.UI.MainMenu {
+    using System.Collections.Generic;
+    using System.Linq;
+    using ColossalFramework.UI;
+    using TrafficManager.U;
+    using TrafficManager.U.Autosize;
+    using UnityEngine;
+
+    public partial class MainMenuWindow {
+        internal class ToolPanel : UPanel {
+            internal struct AddButtonsResult {
+                public List<BaseMenuButton> Buttons;
+                public MainMenuLayout Layout;
+            }
+
+            /// <summary>Create buttons and add them to the given panel UIBuilder.</summary>
+            /// <param name="builder">UI builder to use.</param>
+            /// <param name="atlasKeysSet">Atlas keys to update for button images.</param>
+            /// <param name="buttonDefs">Button defs collection to create from it.</param>
+            /// <param name="minRowLength">Longest the row can be without breaking.</param>
+            /// <returns>A list of created buttons.</returns>
+            private AddButtonsResult AddButtonsFromButtonDefinitions(
```

Comment:
Missing params, `atlasKeysSet` does not exist

Reply:
Updated XMLdoc